solr/core/src/test/org/apache/solr/search/TestFilteredDocIdSet.java

```diff
@@ -127,22 +130,49 @@ public void testNullDocIdSet() throws Exception {
     IndexSearcher searcher = newSearcher(reader);
     Assert.assertEquals(1, searcher.search(new MatchAllDocsQuery(), 10).totalHits.value);
     
-    // Now search w/ a Filter which returns a null DocIdSet
-    Filter f = new Filter() {
+    // Now search w/ a Query which returns a null DocIdSet
+    Query f = new Query() {
       @Override
-      public DocIdSet getDocIdSet(LeafReaderContext context, Bits acceptDocs) {
-        return null;
+      public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float boost) {
+        return new Weight(this) {
+
+          @Override
+          public Explanation explain(LeafReaderContext context, int doc) throws IOException {
+            final Scorer scorer = scorer(context);
+            final boolean match = (scorer != null && scorer.iterator().advance(doc) == doc);
+            if (match) {
+              assert scorer.score() == 0f;
+              return Explanation.match(0f, "Match on id " + doc);
```

Comment:
This code will never be called, should simplify our implementation of `explain` based on knowing that `scorer` always returns `null`. This same comment applies later in the same file and also in TestSort.

Reply:
Yes, I agree. I will have that fixed. Thanks for pointing it out!